File: src/main/java/ar/edu/unq/epers/unidad4/service/interfaces/ItemService.java
package ar.edu.unq.epers.unidad4.service.interfaces;

import ar.edu.unq.epers.unidad4.model.Item;
import ar.edu.unq.epers.unidad4.persistence.sql.entity.ItemSQL;

import java.util.Collection;

public interface ItemService {
    Item guardar(Item itemSQL);
    Item recuperar(Long itemId);
    Collection<Item> getMasPesados(int peso);
    Collection<Item> getItemsDePersonajesDebiles(int vida);
    void clearAll();
}

File: src/main/kotlin/ar/edu/unq/epers/unidad4/service/ItemService.kt
package ar.edu.unq.epers.unidad4.service

import ar.edu.unq.epers.unidad4.model.Item

interface ItemService {
    fun guardar(item: Item): Item
    fun recuperar(itemId: Long): Item
    fun getMasPesados(peso: Int): Collection<Item>
    fun getItemsPersonajesDebiles(vida: Int): Collection<Item>
    fun clearAll()
}